src/main/java/cd/go/contrib/elasticagent/executors/ValidateConfigurationExecutor.java

```diff
@@ -16,55 +16,83 @@
+import cd.go.contrib.elasticagent.requests.ValidatePluginSettingsRequest;
 import com.thoughtworks.go.plugin.api.response.DefaultGoPluginApiResponse;
 import com.thoughtworks.go.plugin.api.response.GoPluginApiResponse;
+import io.fabric8.kubernetes.api.model.Namespace;
+import io.fabric8.kubernetes.client.KubernetesClient;
 
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 import static cd.go.contrib.elasticagent.KubernetesPlugin.LOG;
-import static cd.go.contrib.elasticagent.executors.GetPluginConfigurationExecutor.FIELDS;
-import static cd.go.contrib.elasticagent.executors.GetPluginConfigurationExecutor.GO_SERVER_URL;
+import static cd.go.contrib.elasticagent.executors.GetPluginConfigurationExecutor.*;
 import static cd.go.contrib.elasticagent.utils.Util.GSON;
+import static java.text.MessageFormat.format;
 import static org.apache.commons.lang3.StringUtils.isBlank;
 
 public class ValidateConfigurationExecutor implements RequestExecutor {
-    private final ValidatePluginSettings settings;
+    private final ValidatePluginSettingsRequest validatePluginSettingsRequest;
     private PluginRequest pluginRequest;
+    private final KubernetesClientFactory factory;
     private List<Map<String, String>> result = new ArrayList<>();
 
-    public ValidateConfigurationExecutor(ValidatePluginSettings settings, PluginRequest pluginRequest) {
-        this.settings = settings;
+    public ValidateConfigurationExecutor(ValidatePluginSettingsRequest validatePluginSettingsRequest, PluginRequest pluginRequest) {
+        this(validatePluginSettingsRequest, pluginRequest, KubernetesClientFactory.instance());
+    }
+
+    ValidateConfigurationExecutor(ValidatePluginSettingsRequest validatePluginSettingsRequest, PluginRequest pluginRequest, KubernetesClientFactory factory) {
+        this.validatePluginSettingsRequest = validatePluginSettingsRequest;
         this.pluginRequest = pluginRequest;
+        this.factory = factory;
     }
 
     public GoPluginApiResponse execute() throws ServerRequestFailedException {
         LOG.debug("Validating plugin settings.");
 
         for (Map.Entry<String, Field> entry : FIELDS.entrySet()) {
             Field field = entry.getValue();
-            Map<String, String> validationError = field.validate(settings.get(entry.getKey()));
+            Map<String, String> validationError = field.validate(validatePluginSettingsRequest.get(entry.getKey()));
 
             if (!validationError.isEmpty()) {
                 result.add(validationError);
             }
         }
 
         validateGoServerUrl();
+        validateNamespaceExistence();
 
         return DefaultGoPluginApiResponse.success(GSON.toJson(result));
     }
 
+    private void validateNamespaceExistence() {
+        try {
+            final String namespace = validatePluginSettingsRequest.getPluginSettingsMap().getNamespace();
+            if (isBlank(namespace)) {
```

Comment:
We set the namespace to be `default` in plugin settings. The only chance of this condition passing is if user passes `" "`. In that case, we should validate and error out, right?

Reply:
Sure. Let me check

Reply:
This[https://github.com/gocd/kubernetes-elastic-agents/pull/24/files#diff-b6d3e1e53a7c2cd2538c06fd65cb6a13R105] considers empty string as a null and always returns a default value. I removed blank check for namespace from `validateNamespaceExistence `. WDYT ?